bsp/wavious-mcu/metal/riscv_cpu_min.c

```diff
@@ -0,0 +1,155 @@
+ */
+#include <metal/machine.h>
+#include <metal/drivers/riscv_cpu_min.h>
+
+/*******************************************************************************
+**                   INTERRUPT CONTROLLER VTABLE
+*******************************************************************************/
+static void __metal_driver_riscv_cpu_controller_interrupt_init(struct metal_interrupt *controller)
+{
+    struct __metal_driver_riscv_cpu_min_intc *intc = (void *)(controller);
+    if (!intc->init_done)
+    {
+        // Clear out table
+        for (uint8_t i = 0; i < METAL_MAX_MI; i++)
+        {
+            intc->metal_int_table[i].handler = NULL;
+            intc->metal_int_table[i].sub_int = NULL;
+            intc->metal_int_table[i].exint_data = NULL;
+        }
+    }
+    intc->init_done = 1;
+}
+
+static int __metal_driver_riscv_cpu_controller_interrupt_register(struct metal_interrupt *controller,
+                                                                  int id,
+                                                                  metal_interrupt_handler_t isr,
+                                                                  void *priv)
+{
+    struct __metal_driver_riscv_cpu_min_intc *intc = (void *)(controller);
+
+    if (id >= METAL_MAX_MI)
+    {
+        return -11;
+    }
+
+    intc->metal_int_table[id].handler = isr;
+    intc->metal_int_table[id].exint_data = priv;
+    return 0;
+}
+
+static int __metal_driver_riscv_cpu_controller_interrupt_enable(struct metal_interrupt *controller,
+                                                                int id)
+{
+    uint32_t mask;
+    if (id >= METAL_MAX_MI)
+    {
+        return -1;
+    }
+
+    mask = 1 << id;
+    __asm__ volatile("csrs mie, %0" : : "r" (mask));
+    return 0;
+}
+
+static int __metal_driver_riscv_cpu_controller_interrupt_disable(struct metal_interrupt *controller,
+                                                                 int id)
+{
+    uint32_t mask;
+    if (id >= METAL_MAX_MI)
+    {
+        return -1;
+    }
+
+    mask = 1 << id;
+    __asm__ volatile("csrc mie, %0" : : "r" (mask));
+    return 0;
+}
+
+static metal_vector_mode __metal_driver_riscv_cpu_controller_get_vector_mode(struct metal_interrupt *controller)
+{
+    uintptr_t val;
+    __asm__ volatile("csrr %0, mtvec" : "=r"(val));
+    val &= 0x1;
+
+    if (val == METAL_MTVEC_VECTORED)
+    {
+        return METAL_VECTOR_MODE;
+    }
+
+    return METAL_DIRECT_MODE;
+}
+
+static int __metal_driver_riscv_cpu_controller_set_vector_mode(struct metal_interrupt *controller,
+                                                               metal_vector_mode mode)
+{
+    // Not supported
+    return -1;
+}
+
+static int __metal_driver_riscv_cpu_controller_command_request(struct metal_interrupt *controller,
+                                                               int cmd,
+                                                               void *data)
+{
+    // Not supported
+    return 0;
```

Comment:
Should this return -1 as well as it isn't supported? or does that break something?

Reply:
This is how it was implemented in their source file riscv_cpu.c so I made it the same. It technically isn't used so I think it's okay.